pyiceberg/expressions/__init__.py

```diff
@@ -725,12 +729,37 @@ def as_bound(self) -> Type[BoundNotIn[L]]:
         return BoundNotIn[L]
 
 
-class LiteralPredicate(UnboundPredicate[L], ABC):
-    literal: Literal[L]
+class LiteralPredicate(IcebergBaseModel, UnboundPredicate[L], ABC):
+    op: str = Field(
+        default="",
+        alias="type",
+        validation_alias="type",
+        serialization_alias="type",
+        repr=False,
+    )
```

Comment:
Why not call it `type`:
```suggestion
    type: TypingLiteral["lt-eq", "gt", "gt-eq", "eq", "not-eq", "starts-with", "not-starts-with"] = Field(alias="type")
```

Reply:
If you call it directly type, you cannot subclass it directly and use:
- For the father: type: TypingLiteral["lt-eq", "gt", "gt-eq", "eq", "not-eq", "starts-with", "not-starts-with"] = Field(alias="type")
- For the child: type = type: TypingLiteral["eq"] = Field(default="eq")

Because mypy will scream at you saying:

```
1. "type" overrides symbol of same name in class "LiteralPredicate"
     Variable is mutable so its type is invariant
       Override type "Literal['eq']" is not the same as base type "Literal['lt-eq', 'gt', 'gt-eq', 'eq', 'not-eq', 'starts-with', 'not-starts-with']" [reportIncompatibleVariableOverride]
```